src/mp_game_settings.hpp

```diff
@@ -85,8 +85,10 @@ struct mp_game_settings
 	{
 		boost::optional<version_info> version;
 		boost::optional<version_info> min_version;
-
 		std::string name;
+		std::string type;
+		bool required;
+		std::string content_id;
```

Comment:
An `addon_version_info` is created once per scenario/era/modification though, not once per add-on.  So the name is somewhat misleading, but otherwise I think this is correct?

Reply:
no, the code inside update_addon_requirements merges all addon_version_info of a specific addon into one. while keeping the min_version oto the minimum of all min_versions.

Reply:
hm.  I guess those new `addon_version_info` attributes would need to be comma-delimited lists instead then, one item per piece of content in the add-on?  Though then that would still raise the same issues as the previous implementation with needing to check for missing or malicious data in the list.  Or maybe it would make more sense to send a separate [multiplayer][content] set of configs to provide this additional data on all content being used?  Or maybe add a set of [content] configs within each [multiplayer][addon] config?

Reply:
>  Though then that would still raise the same issues as the previous implementation with needing to check for missing or malicious data in the list

I'm not sure what you mean here

Reply:
The 1.14 implementation needs to verify that the comma delimited lists are all the same length, which might not be the case either for older clients that don't send the source add-on and add-on version attributes or in the case of someone intentionally changing their client to not send the attributes.  The former caused issues with the original 1.14 implementation (trying to access an index of the vector returned by utils::split that doesn't exist).

Trying to access an attribute that doesn't exist on a config doesn't have that problem, since it would just return the default value (in this case an empty string).